index.js

```diff
@@ -160,10 +165,20 @@ Filter.prototype.build = function() {
             return fs.unlinkSync(outputPath);
           } case 'change': {
             instrumentation.change++;
-            return plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            var changeOperation = plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            if (plugin.async) {
```

Comment:
this seems like a general throttling problem, the concurrent versions will also run into.

Where should the throttling happen?